src/DemoFile/DemoParser.cs

```diff
@@ -62,7 +67,7 @@ protected DemoParser()
     public ref BaseUserMessageEvents BaseUserMessageEvents => ref _baseUserMessageEvents;
     public ref TempEntityEvents TempEntityEvents => ref _tempEntityEvents;
 
-    public CDemoFileHeader FileHeader { get; private set; } = new();
+    public CDemoFileHeader? FileHeader { get; private set; }
```

Comment:
it doesn't make sense to initialize this to invalid values. Also, the message has not been read yet.